src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -293,10 +293,14 @@ public void close(Collection<TopicPartition> partitions) {
               partitionChannelKey(topicPartition.topic(), topicPartition.partition());
           TopicPartitionChannel topicPartitionChannel =
               partitionsToChannel.get(partitionChannelKey);
-          topicPartitionChannel.closeChannel();
+          // Check for null since it's possible that the something goes wrong even before the
```

Comment:
When can this happen?

Reply:
It could happen in many cases, for example, if the table doesn't have a correct schema, or we don't have permission to ALTER COLUMN, see `createTableIfNotExists ` in `SnowflakeSinkServiceV2.startTask` if you're interested